src/components/InternalLink/index.tsx

```diff
@@ -28,6 +31,7 @@ export default function InternalLink({ href, children }) {
           href = `${hrefParts[0]}/q/${filterKind}/${filterKey}#${hrefParts[1]}`;
         } else {
           href += `/q/${filterKind}/${filterKey}`;
+          useEffect(() => setUrl(href), []);
```

Comment:
does this need a `useEffect` to set the href properly? Can all of this conditional logic be inside of the `useEffect`? https://legacy.reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

parsing local storage (i.e. from `window`) should also reside inside the useEffect as `window` will not be available server-side  (during prerendering)

Reply:
This is hydration related. I missed removing it when separating out the hydration efforts. Will address this on that branch. Thanks